system/HTTP/Negotiate.php

```diff
@@ -189,6 +194,62 @@ protected function getBestMatch(
         return $strictMatch ? '' : $supported[0];
     }
 
+    /**
+     * Strict locale search, including territories (en-*)
+     *
+     * @param list<string> $supported App-supported values
+     * @param ?string      $header    Compatible 'Accept-Language' header string
+     */
+    protected function getBestLocaleMatch(array $supported, ?string $header): string
```

Comment:
It may be better to use nouns here.
```suggestion
    protected function getBestLocaleMatch(array $supportedLocales, ?string $header): string
```